pkg/daemon/daemon.go

```diff
@@ -963,8 +980,105 @@ func (dn *Daemon) applyHaProfiles(nodeProfile *ptpv1.PtpProfile, cmdLine string)
 				}
 			}
 		}
-		cmdLine = fmt.Sprintf("%s %s", cmdLine, strings.Join(updateHaProfileToSocketPath, " "))
+		cmdLine = fmt.Sprintf("%s%s", cmdLine, strings.Join(updateHaProfileToSocketPath, " "))
 	}
 	glog.Infof(cmdLine)
 	return cmdLine
 }
+
+func listHaProfiles(nodeProfile *ptpv1.PtpProfile) (haProfiles []string) {
+	if profiles, ok := nodeProfile.PtpSettings[PTP_HA_IDENTIFIER]; ok {
+		haProfiles = strings.Split(profiles, ",")
+		return
+	}
+	return
+}
+
+func (dn *Daemon) getHaProfiles(nodeProfile *ptpv1.PtpProfile) (haProfiles map[string][]string) {
+	listProfiles := listHaProfiles(nodeProfile)
+	haProfiles = make(map[string][]string, len(listProfiles))
+	for _, profileName := range listProfiles {
+		if strings.TrimSpace(profileName) != "" {
+			for _, dmProcess := range dn.processManager.process { // look for profiles to get their interfaces list
+				if strings.TrimSpace(*dmProcess.nodeProfile.Name) == strings.TrimSpace(profileName) {
+					var ifaces []string
```

Comment:
could also get updateHaProfileToSocketPath here, and return cmdLine with this function, so that function applyHaProfiles() is no longer needed which was quite repetitive.

Reply:
fixed it